lib/task.js

```diff
@@ -309,17 +309,24 @@ function factory(
         var self = this;
 
         self.instantiateJob();
+        // Support schedulerOverrides for backwards graph definition compatibility
+        var timeout = self.definition.options.timeout;
+        if (!timeout && self.definition.options.schedulerOverrides) {
+            timeout = self.definition.options.schedulerOverrides.timeout;
+        }
```

Comment:
- How about setting a default timeout because we don't want a task to be hang forever? I remembered previously it is 24 hours.
- How about if the timeout is not a number? I see you removed the assertion.number for the timeout, what's the reason? Though the `undefined` works, but how about a string/an object or others